server/asset/eth/coiner.go

```diff
@@ -284,3 +286,203 @@ func (c *swapCoin) Value() uint64 {
+	if err != nil {
+		return 0, 0, err
+	}
+	bigBal, err := backend.node.balance(backend.rpcCtx, addr)
+	if err != nil {
+		return 0, 0, err
+	}
+	bal, err := ToGwei(bigBal)
+	if err != nil {
+		return 0, 0, err
+	}
+	return bal, pendingBal, nil
+}
+
+// lowestGasPrice returns the lowest gas fee among pending transactions or zero.
+func (backend *Backend) lowestGasPrice(addr *common.Address) (uint64, error) {
+	txsMap, err := backend.node.txpoolContent(backend.rpcCtx)
+	if err != nil {
+		return 0, fmt.Errorf("unable to get txpool content: %v", err)
+	}
+	var lowestPrice *big.Int
+	for from, txs := range txsMap["pending"] {
+		if from == addr.String() {
+			// Skip debits.
+			continue
+		}
+		for _, tx := range txs {
+			if *tx.To() == *addr {
+				if lowestPrice == nil {
+					lowestPrice = tx.GasPrice()
+					continue
+				}
+				if lowestPrice.Cmp(tx.GasPrice()) > 0 {
+					lowestPrice = tx.GasPrice()
+				}
+			}
+		}
+	}
+	// If we didn't find a pending tx, the outstanding balance was likely
+	// mined before we checked so no error.
+	if lowestPrice == nil {
+		return 0, nil
+	}
+	lp, err := ToGwei(lowestPrice)
+	if err != nil {
+		return 0, fmt.Errorf("unable to convert pending fee rate to gwei: %v", err)
+	}
+	return lp, nil
+}
+
+// Confirmations returns one if an account currently has the funds to satisfy
+// its funding amount.
+func (c *amountCoin) Confirmations(_ context.Context) (int64, error) {
+	bal, pendingBal, err := c.backend.accountBalance(&c.cID.Address)
+	if err != nil {
+		return -1, err
+	}
+	amt := c.cID.Amount
+	switch {
+	case bal >= amt:
+		// While not completely accurate, node.balance will only
+		// return funds that have been mined, and therefore have at
+		// least one confirmation.
+		return 1, nil
+	case pendingBal >= amt:
+		// The pending balance was found to be enough.
+		return 0, nil
+	}
+	return -1, fmt.Errorf("account %s does not have %d gwei", &c.cID.Address, amt)
+}
+
+// ID is the amount's coin ID.
+func (c *amountCoin) ID() []byte {
+	return c.cID.Encode()
+}
+
+// There may be more than TxID leading to an amountCoin, and no way to know
+// which transactions led to this state beyond a certain window without an
+// archival node.
+func (c *amountCoin) TxID() string {
+	return ""
+}
+
+// String is a human readable representation of the amount coin.
+func (c *amountCoin) String() string {
+	return c.cID.String()
+}
+
+// Value is the amount held in this amountCoin.
+func (c *amountCoin) Value() uint64 {
+	return c.cID.Amount
+}
+
+// FeeRate returns the lowest gas fee among pending transactions to the
+// account. It is not set if there was not pending balance upon initiation of
+// the funding coin.
+func (c *amountCoin) FeeRate() uint64 {
+	return c.gasPrice
+}
+
+// Auth ensures that a user really has the keys to spend a funding coin by
+// verifying a message signed by the funding coin address.
+func (c *amountCoin) Auth(pubkeys, sigs [][]byte, msg []byte) error {
```

Comment:
I guess we'll still have to figure out when and how we confirm ownership of the account, but it won't be via coin like this anymore.

Reply:
Yeah, this pr is only the last commit, so I'll rebase and remove that.